### ISSUE TITLE
Enhancement: How to fix @typescript-eslint/restrict-template-expressions issues in .js files

### ISSUE THREAD

**Issue by da-du** [typescript-eslint/typescript-eslint#7029]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/restrict-template-expressions/

### Description

This is a .js file, except disable, how to fix this kind of problem
<img width="1172" alt="image" src="https://github.com/typescript-eslint/typescript-eslint/assets/106589025/964ad5f9-55c0-42df-9650-7da2d963e5ad">

### Fail

```typescript
renderStyle(src) {
    const style = {};
    if (src) {
        /**
         * Invalid type "any" of template literal expression.eslint@typescript-eslint/restrict-template-expressions
         * (parameter) src: any
         */
        style.background = `url(${src}) no-repeat`;
        style.backgroundSize = 'cover';
    }

    return style;
}
```

### Pass

```typescript
renderStyle(src) {
    const style = {};
    if (src) {
        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
        style.background = `url(${src}) no-repeat`;
        style.backgroundSize = 'cover';
    }

    return style;
}
```

**Comment by armano2** [comment]
this is working as intended, you are missing type of `src`

```
renderStyle(src: THIS_IS_ANY) {
```

**Comment by armano2** [comment]
https://v6--typescript-eslint.netlify.app/play#ts=5.0.2&fileType=.js&code=MYewdgzgLgBAHjAvDA3gWAFAxgegFR4wACADgIYBOZAtqtBQJZgDmAvjBBcDHjpthQCmYACaCKAZSgBPADaCAFJ2ABKVP2wxQkWNDmCkqVgG4N2BgDMYSrmvRZN2PfIB0AIzLAA1swogArqKGAAb%2BFLIKACQoyqxqYCAAtEIkgmRQwaYOjs6C7p4%2BfoEiEgwAXgbIAOSgAG7iVVmOrJhmMEJQYWAcMvJNMC0YrEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aFfKsu-RAW2LwAhn3oAPYh2TJKAeyaoMkRNGhzokcGAC%2BIHUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

```js
  /** @param {string} src */
  renderStyle(src) {
    const style = {};
    if (src) {
      style.background = `url(${src}) no-repeat`;
      style.backgroundSize = 'cover';
    }

    return style;
  }
```

**Comment by bradzacher** [comment]
Armano's point 2 is also mentioned in our FAQ https://typescript-eslint.io/linting/troubleshooting#one-of-my-lint-rules-isnt-working-correctly-on-a-pure-javascript-file

The rules are powered by types - so you need to provide types or else TS will infer `any`.
